libdevcore/CommonData.h

```diff
@@ -120,6 +120,134 @@ inline void toBigEndian(T _val, Out& o_out)
 		o_out[i - 1] = (typename Out::value_type)(uint8_t)v;
 	}
 }
+#ifdef __INTEL_COMPILER
```

Comment:
the implicit conversion used in the template function does not compile using intel platform.
When the argument type is specified, we can call the convert_to member function.
In summary, the intel platform supports this "convert_to" member function, but does not support implicit conversion.
This member function is not available with the template parameter.